Automattic/wp-calypso
# HelpSearch: migrate to use `react-query`

#### Changes proposed in this Pull Request

* `HelpSearch`: migrate to use `react-query` to fetch help links instead of data layer + redux
* Create a new `useHelpSearchQuery` and migrate `useInlineHelpSearchQuery` to use that as a base

#### Testing instructions

* Go to `/help` and perform some searches, make sure they work as expected
* Also test the inline help search and make sure it still works as expected

Note: code style errors are preexisting
